src/session.ts

```diff
@@ -43,21 +43,13 @@ let activeBrowserExternalUri: Uri;
 export function deploySessionWatcher(extensionPath: string): void {
     console.info(`[deploySessionWatcher] extensionPath: ${extensionPath}`);
     resDir = path.join(extensionPath, 'dist', 'resources');
-    watcherDir = path.join(os.homedir(), '.vscode-R');
+    const tempPath = os.tmpdir();
+    watcherDir = path.join(tempPath, '.vscode-R');
```

Comment:
We put the directory in user home because I want to ensure that the directory and files in it are private to the user. Does putting in the temp directory also guarantee that?

Also, on a multi-user server, the path will conflict with other users if they also use the extension.

AFAIK, there's also practice putting some files in user home in other extensions for similar purposes.

I'm not seeing a clear advantage of putting files in system temp folder here.

Reply:
On my windows machine, the temp dir is private for each user, so I didn't consider this limitation.
Considering this, I guess it makes sense to leave these files where they are.

Reply:
On a linux machine, the if we create a folder in temp, it does not seem to make it `700` private to user by default. I guess we have to put it in home folder to be secure for all platforms.

Reply:
Agreed